### PR TITLE
Add a tag section

### LINKED ISSUE #35: Add tags to the resources and add a section to filter resources by tags
## Problem

Currently, there is no tag to differentiate the resources as articles, blogs, news, workshop, webinar, tutorial or FAQs.

## Description

- We can add tags to the resources and also add a section on the left side to filter the resources based on the tags chosen by the user.
- By default, all the resources will be shown
- A panel with tags and checkboxes can be added on the left side from where tags can be chosen.

## Implementation

- [x] I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by tinniaru3005**
## Fixes

Fixes #35 by @tinniaru3005

## Description

Add a tag section for filtering resources based on All, Articles, Pamphlets, Course, Poster, Video

## Technical details

Currently only the HTML, CSS is completed. Will be adding the JavaSCript part to add the functionality for filtering the resources

## Screenshots

cc-res

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by tinniaru3005** [comment]
@possumbilities please let me know if the design looks good to you. Then I will go ahead and start working on adding the JavaSCript code as well so as to add the functionality to filter the resources.

**Comment by possumbilities** [comment]
@tinniaru3005 Hi! I like the direction, looking forward to the JS functionality! ❤️

**Comment by tinniaru3005** [comment]
Sure @possumbilities I will start working on the JS functionality today. I am glad you liked the initial design. 🤩

File: docs/index.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Get all tag list items
+const tagItems = document.querySelectorAll('.tag-section li');
+
+// Loop through the tag list items
+for (let i = 0; i < tagItems.length; i++) {
+  // Add a click event listener to each tag item
+  tagItems[i].addEventListener('click', function() {
+    // Remove the 'active' class from all tag items
```
Comment (Murdock9803) [inline docs/index.js:8]: @tinniaru3005 Removing the active class from all tag items, each time an item is clicked is unnecessary and makes the code less optimized. Instead, we can just remove the active class from previously tagged item.

**Comment by Murdock9803** [comment]
@possumbilities 
I have studied the code and planned the changes. I am also planning to extend this feature, and convert the present filtering mechanism (`topic` `medium` `language`) to an intuitive UI based approach, So that users can select their resources with ease. 

But I have already made PR #204  and #188 , which are not merged yet and it will create merge conflicts if i raise another PR as the files are coinciding. So, please have a look at those PRs so that I can add the required functionality.
I am really excited to implement the feature.

**Comment by possumbilities** [comment]
I've made a new branch that pulls in the latest cleanup from Timid Robot called `tag-section`. https://github.com/creativecommons/cc-resource-archive/tree/tags-section I would suggest any future work can build from this branch and then open a PR to merge in the completed work from there.

**Pull request closed without merging**